### ISSUE TITLE
Using createServer.listen() doesn't correctly detect used ports

### ISSUE THREAD

**Issue by zen0wu** [argos-ci/jest-puppeteer#281]
## 🐛 Bug Report

`getIsPortTaken` function doesn't seem to work correctly under macOS.

## To Reproduce

1. `nc -l localhost 3000` to start a minimal server locally
2. Start  `jest-dev-server` on the same port with usedPortAction set to "ask"
3. It doesn't ask

## Expected behavior

It should ask

## Link to repl or repo (highly encouraged)

Please provide a minimal repository on GitHub.

Issues without a reproduction link are likely to stall.

## Run `npx envinfo --system --binaries --npmPackages expect-puppeteer,jest-dev-server,jest-environment-puppeteer,jest-puppeteer,spawnd --markdown --clipboard`

Paste the results here:

```bash
*** Clipboard option removed - use clipboardy or clipboard-cli directly ***


## System:
 - OS: macOS 10.14.6
 - CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
 - Memory: 1.98 GB / 16.00 GB
 - Shell: 5.7.1 - /usr/local/bin/zsh
## Binaries:
 - Node: 12.10.0 - /usr/local/bin/node
 - Yarn: 1.17.3 - /usr/local/bin/yarn
 - npm: 6.11.3 - /usr/local/bin/npm
 - Watchman: 4.9.0 - /usr/local/bin/watchman
## npmPackages:
 - jest-dev-server: ^4.3.0 => 4.3.0
```

**Comment by domoritz** [comment]
I'm running into this issue. This bug breaks Jest in VSCode since the local server keeps running and different runs fail all the time.

**Comment by BitFis** [comment]
Same issue here, this still seems to be broken

**Comment by gregberge** [comment]
Fixed in #518
